llms/mlx_lm/utils.py

```diff
@@ -130,10 +130,11 @@ def generate(
 
         if verbose:
             s = tokenizer.decode(tokens)
-            print(s[skip:], end="", flush=True)
-            skip = len(s)
+            if '�' not in s:
```

Comment:
The fix makes sense to me. However, should we make the special character a constant instead of a magic value? It would make the code more readable. Maybe ?
```
REPLACEMENT_CHAR = '\ufffd'

       s = tokenizer.decode(tokens)
        if REPLACEMENT_CHAR not in s:
            print(s[skip:], end="", flush=True)
            skip = len(s)

tokens = tokenizer.decode(tokens).replace(REPLACEMENT_CHAR, '')
   ```

Reply:
I have adopted your suggestion and updated this PR.

Reply:
Good suggestion, thanks @mzbac